### ISSUE TITLE
Log scale on pcolor plot with only one tick

### ISSUE THREAD

**Issue by samcunliffe** [matplotlib/matplotlib#8557]
### Bug report

**Bug summary**

The color scale only has one tickmark. This is never correct.

**Code for reproduction**

```python
import numpy as np
import matplotlib
matplotlib.use('pdf')
import matplotlib.pyplot as plt
from matplotlib.colors import LogNorm

gcmap = 'plasma'

# random data
mi = 0.00058265
ma = 0.00108149

X, Y = np.meshgrid(range(512), np.linspace(0, 0.3 * 199, 200))
Z = mi+(ma-mi)*np.random.random(X.shape)

# log z axis
plt.pcolor(Z, cmap=gcmap,
           norm=LogNorm(vmin=abs(Z.min()), vmax=abs(Z.max())))
plt.colorbar()
plt.xlim(0, 512)
plt.savefig("plot.pdf")
```

**Actual outcome**
[plot.pdf](https://github.com/matplotlib/matplotlib/files/968515/plot.pdf)

**Expected outcome**
A colorbar with at least two tickmarks.

**Matplotlib version**
  * Operating System: Mac OS X 10.12.4
  * Matplotlib Version: 1.5.2
  * Python Version: 3.5.2

installed via pip3, but reproducible on other systems

**Comment by jstrube** [comment]
Looks like this is fixed in v2.0.0
[plot.pdf](https://github.com/matplotlib/matplotlib/files/968537/plot.pdf)

**Comment by tacaswell** [comment]
Closing as already fixed.  If this is not the case please ping to have this reopened.
